<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>ChIPexoQual: A quality control pipeline for ChIP-exo/nexus data. &bull; ChIPexoQual</title><!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous"><script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous"><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ChIPexoQual</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>ChIPexoQual: A quality control pipeline for ChIP-exo/nexus data.</h1>
                        <h4 class="author"></h4><p>Rene Welch and S&uuml;nd&uuml;z Kele&#351;<br>
Department of Statistics, University of Wisconsin-Madison</p>
            
          </div>

    
    
<div class="contents">
<p><meta http-equiv="content-type" content="text/html;charset=utf-8"></p>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="overview" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#overview" class="anchor"> </a></body></html>Overview</h1>
<p>In this vignette, we provide a brief overview of the <em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em> package. This package provides a statistical quality control (QC) pipeline that enables the exploration and analysis of ChIP-exo/nexus experiments. In this vignette we used the reads aligned to <strong>chr1</strong> in the mouse liver ChIP-exo experiment <span class="citation">(Serandour et al. 2013)</span> to illustrate the use of the pipeline. To load the packages we use:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">library</span>(ChIPexoQual)
    <span class="kw">library</span>(ChIPexoQualExample)</code></pre></div>
<p><em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em> takes a set of aligned reads from a ChIP-exo (or ChIP-nexus) experiment as input and performs the following steps:</p>
<ol style="list-style-type: decimal"><li>Identify read islands, i.e., overlapping clusters of reads separated by gaps, from read coverage.</li>
<li>Compute <span class="math inline">\(D_i\)</span>, number of reads in island <span class="math inline">\(i\)</span>, and <span class="math inline">\(U_i\)</span>, number of island <span class="math inline">\(i\)</span> positions with at least one aligning read, <span class="math inline">\(i=1, \cdots, I\)</span>.
<ul><li>For each island <span class="math inline">\(i\)</span>, <span class="math inline">\(i=1, \cdots, I\)</span> compute island statistics: <span class="math display">\[ 
 \begin{align*}
\mbox{ARC}_i &amp;= \frac{D_i}{W_i}, \quad \mbox{URC}_i = \frac{U_i}{D_i},  \\
%\mbox{URC}_i &amp;= \frac{U_i}{D_i}, \\
\mbox{FSR}_i &amp;= \frac{(\text{Number of forward strand reads aligning to
  island $i$})}{D_i},
 \end{align*} 
\]</span></li>
</ul>
where <span class="math inline">\(W_i\)</span> denotes the width of island <span class="math inline">\(i\)</span>,.</li>
<li>Generate diagnostic plots (i) URC vs.&nbsp;ARC plot; (ii) Region Composition plot; (iii) FSR distribution plot.</li>
<li>Randomly sample <span class="math inline">\(M\)</span> (at least 1000) islands and fit, <span class="math display">\[    
\begin{align*}
D_i = \beta_1 U_i + \beta_2 W_i + \varepsilon_i,
\end{align*}
\]</span> where <span class="math inline">\(\varepsilon_i\)</span> denotes the independent error term. Repeat this process <span class="math inline">\(B\)</span> times and generate box plots of estimated <span class="math inline">\(\beta_1\)</span> and <span class="math inline">\(\beta_2\)</span>.</li>
</ol><p>We analyzed a larger collection of ChIP-exo/nexus experiments in <span class="citation">(Welch et al. 2016)</span> including complete versions of this samples.</p>
</div>
<div id="creating-an-exodata-object" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#creating-an-exodata-object" class="anchor"> </a></body></html>Creating an <strong>ExoData</strong> object</h1>
<p>The minimum input to use <em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em> are the aligned reads of a ChIP-exo/nexus experiment. <em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em> accepts either the name of the bam file or the reads in a <em>GAlignments</em> object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    files =<span class="st"> </span><span class="kw">list.files</span>(<span class="kw">system.file</span>(<span class="st">"extdata"</span>,
        <span class="dt">package =</span> <span class="st">"ChIPexoQualExample"</span>),<span class="dt">full.names =</span> <span class="ot">TRUE</span>)
    <span class="kw">basename</span>(files[<span class="dv">1</span>])</code></pre></div>
<pre><code>## [1] "ChIPexo_carroll_FoxA1_mouse_rep1_chr1.bam"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    ex1 =<span class="st"> </span><span class="kw"><a href="../reference/ExoData-class.html">ExoData</a></span>(<span class="dt">file =</span> files[<span class="dv">1</span>],<span class="dt">mc.cores =</span> 2L,<span class="dt">verbose =</span> <span class="ot">FALSE</span>)
    ex1</code></pre></div>
<pre><code>## ExoData object with 655785 ranges and 11 metadata columns:
##            seqnames                 ranges strand |  fwdReads  revReads
##               &lt;Rle&gt;              &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;
##        [1]     chr1     [3000941, 3000976]      * |         2         0
##        [2]     chr1     [3001457, 3001492]      * |         0         1
##        [3]     chr1     [3001583, 3001618]      * |         0         2
##        [4]     chr1     [3001647, 3001682]      * |         1         0
##        [5]     chr1     [3001852, 3001887]      * |         1         0
##        ...      ...                    ...    ... .       ...       ...
##   [655781]     chr1 [197192012, 197192047]      * |         0         1
##   [655782]     chr1 [197192421, 197192456]      * |         0         1
##   [655783]     chr1 [197193059, 197193094]      * |         1         0
##   [655784]     chr1 [197193694, 197193729]      * |         0         3
##   [655785]     chr1 [197194986, 197195021]      * |         0         2
##               fwdPos    revPos     depth uniquePos                ARC
##            &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;          &lt;numeric&gt;
##        [1]         1         0         2         1 0.0555555555555556
##        [2]         0         1         1         1 0.0277777777777778
##        [3]         0         1         2         1 0.0555555555555556
##        [4]         1         0         1         1 0.0277777777777778
##        [5]         1         0         1         1 0.0277777777777778
##        ...       ...       ...       ...       ...                ...
##   [655781]         0         1         1         1 0.0277777777777778
##   [655782]         0         1         1         1 0.0277777777777778
##   [655783]         1         0         1         1 0.0277777777777778
##   [655784]         0         1         3         1 0.0833333333333333
##   [655785]         0         1         2         1 0.0555555555555556
##                          URC       FSR         M         A
##                    &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##        [1]               0.5         1      -Inf       Inf
##        [2]                 1         0      -Inf      -Inf
##        [3]               0.5         0      -Inf      -Inf
##        [4]                 1         1      -Inf       Inf
##        [5]                 1         1      -Inf       Inf
##        ...               ...       ...       ...       ...
##   [655781]                 1         0      -Inf      -Inf
##   [655782]                 1         0      -Inf      -Inf
##   [655783]                 1         1      -Inf       Inf
##   [655784] 0.333333333333333         0      -Inf      -Inf
##   [655785]               0.5         0      -Inf      -Inf
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    reads =<span class="st"> </span><span class="kw">readGAlignments</span>(files[<span class="dv">1</span>],<span class="dt">param =</span> <span class="ot">NULL</span>)
    ex2 =<span class="st"> </span><span class="kw"><a href="../reference/ExoData-class.html">ExoData</a></span>(<span class="dt">reads =</span> reads,<span class="dt">mc.cores =</span> 2L,<span class="dt">verbose =</span> <span class="ot">FALSE</span>)
    <span class="kw">identical</span>(<span class="kw">GRanges</span>(ex1),<span class="kw">GRanges</span>(ex2))</code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>For the rest of the vignette, we generate an <strong>ExoData</strong> object for each replicate:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    files =<span class="st"> </span>files[<span class="kw">grep</span>(<span class="st">"bai"</span>,files,<span class="dt">invert =</span> <span class="ot">TRUE</span>)] ## ignore index files
    exampleExoData =<span class="st"> </span><span class="kw">lapply</span>(files,ExoData,<span class="dt">mc.cores =</span> 2L,<span class="dt">verbose =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>Finally, we can recover the number of reads that compose a <strong>ExoData</strong> object by using the <strong>nreads</strong> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">sapply</span>(exampleExoData,nreads)</code></pre></div>
<pre><code>## [1] 1654985 1766665 1670117</code></pre>
<div id="enrichment-analysis-and-library-complexity" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#enrichment-analysis-and-library-complexity" class="anchor"> </a></body></html>Enrichment analysis and library complexity:</h2>
<p>To create the <em>ARC vs URC plot</em> proposed in <span class="citation">(Welch et al. 2016)</span>, we use the <strong>ARC_URC_plot</strong> function. This function allows to visually compare different samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw"><a href="../reference/ARCvURCplot.html">ARCvURCplot</a></span>(exampleExoData,<span class="dt">names.input =</span> <span class="kw">paste</span>(<span class="st">"Rep"</span>,<span class="dv">1</span>:<span class="dv">3</span>,<span class="dt">sep =</span> <span class="st">"-"</span>))</code></pre></div>
<p><img src="vignette_files/figure-html/ARC1-1.png" width="864" style="display: block; margin: auto;"></p>
<p>This plot typically exhibits one of the following three patterns for any given sample. In all three panels we can observe two arms: the first with low <strong>Average Read Coefficient (ARC)</strong> and varying <strong>Unique Read Coefficient (URC)</strong>; and the second where the <strong>URC</strong> decreases as the <strong>ARC</strong> increases. The first and third replicates exhibit a defined decreasing trend in URC as the ARC increases. This indicates that these samples exhibit a higher ChIP enrichment than the second replicate. On the other hand, the overall URC level from the first two replicates is higher than that of the third replicate, elucidating that the libraries for the first two replicates are more complex than that of the third replicate.</p>
</div>
<div id="strand-imbalance" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#strand-imbalance" class="anchor"> </a></body></html>Strand imbalance</h2>
<p>To create the <em>FSR distribution</em> and <em>Region Composition</em> plots suggested in Welch et. al 2016 (submitted), we use the <strong>FSR_dist_plot</strong> and <strong>region_comp_plot</strong>, respectively.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    p1 =<span class="st"> </span><span class="kw"><a href="../reference/regionCompplot.html">regionCompplot</a></span>(exampleExoData,<span class="dt">names.input =</span> <span class="kw">paste</span>(<span class="st">"Rep"</span>,<span class="dv">1</span>:<span class="dv">3</span>,
        <span class="dt">sep =</span> <span class="st">"-"</span>),<span class="dt">depth.values =</span> <span class="kw">seq_len</span>(<span class="dv">50</span>))
    p2 =<span class="st"> </span><span class="kw"><a href="../reference/FSRDistplot.html">FSRDistplot</a></span>(exampleExoData,<span class="dt">names.input =</span> <span class="kw">paste</span>(<span class="st">"Rep"</span>,<span class="dv">1</span>:<span class="dv">3</span>,<span class="dt">sep =</span> <span class="st">"-"</span>),
        <span class="dt">quantiles =</span> <span class="kw">c</span>(.<span class="dv">25</span>,.<span class="dv">5</span>,.<span class="dv">75</span>),<span class="dt">depth.values =</span> <span class="kw">seq_len</span>(<span class="dv">100</span>))
    gridExtra::<span class="kw">grid.arrange</span>(p1,p2,<span class="dt">nrow =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/strand1-1.png" width="672" style="display: block; margin: auto;"></p>
<p>The left panel displays the <em>Region Composition plot</em> and the right panel shows the <em>Forward Strand Ratio (FSR) distribution plot</em>, both of which highlight specific problems with replicates 2 and 3. The <em>Region Composition plot</em> exhibits apparent decreasing trends in the proportions of regions formed by fragments in one exclusive strand. High quality experiments tend to show exponential decay in the proportion of single stranded regions, while for the lower quality experiments, the trend may be linear or even constant. The FSR distributions of both of replicates 2 and 3 are more spread around their respective medians. The rate at which the FSR distribution becomes more centralized around the median indicates the aforementioned lower enrichment in the second replicate and the low complexity in the third one. The asymmetric behavior of the second replicate is characteristic of low enrichment, while the constant values of replicate three for low minimum number of reads indicate that this replicate has islands composed of reads aligned to very few unique positions.</p>
<div id="further-exploration-of-chip-exo-data" class="section level3">
<h3 class="hasAnchor"><html><body><a href="#further-exploration-of-chip-exo-data" class="anchor"> </a></body></html>Further exploration of ChIP-exo data</h3>
<p>All the plot functions in <em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em> allow a list or several separate <strong>ExoData</strong> objects. This allows to explore island subsets for each replicate. For example, to show that the first arm is composed of regions formed by reads aligned to few positions, we can generate the following plot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw"><a href="../reference/ARCvURCplot.html">ARCvURCplot</a></span>(exampleExoData[[<span class="dv">1</span>]],
                 <span class="kw">subset</span>(exampleExoData[[<span class="dv">1</span>]],uniquePos &gt;<span class="st"> </span><span class="dv">10</span>),
                 <span class="kw">subset</span>(exampleExoData[[<span class="dv">1</span>]],uniquePos &gt;<span class="st"> </span><span class="dv">20</span>),
                 <span class="dt">names.input =</span> <span class="kw">c</span>(<span class="st">"All"</span>, <span class="st">"uniquePos &gt; 10"</span>, <span class="st">"uniquePos &gt; 20"</span>))</code></pre></div>
<p><img src="vignette_files/figure-html/ARC2-1.png" width="864" style="display: block; margin: auto;"></p>
<p>For this figure, we used the <em>ARC vs URC</em> plot to show how several of the regions with low <strong>ARC</strong> values are composed by reads that align to a small number of unique positions. This technique highlights a strategy that can be followed to further explore the data, as with all the previously listed plotting functions we may compare different subsets of the islands in the partition.</p>
</div>
</div>
<div id="quality-evaluation" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#quality-evaluation" class="anchor"> </a></body></html>Quality evaluation</h2>
<p>The last step of the quality control pipeline is to evaluate the linear model:</p>
<p><span class="math display">\[
\begin{align*}
D_i = \beta_1 U_i + \beta_2 U_2 + \epsilon_i,
\end{align*}
\]</span></p>
<p>The distribution of the parameters of this model is built by sampling <strong>nregions</strong> regions (the default value is 1,000), fitting the model and repeating the process <strong>ntimes</strong> (the default value is 100). We visualize the distributions of the parameters with box-plots:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    p1 =<span class="st"> </span><span class="kw"><a href="../reference/paramDistBoxplot.html">paramDistBoxplot</a></span>(exampleExoData,<span class="dt">which.param =</span> <span class="st">"beta1"</span>, <span class="dt">names.input =</span> <span class="kw">paste</span>(<span class="st">"Rep"</span>,<span class="dv">1</span>:<span class="dv">3</span>,<span class="dt">sep =</span> <span class="st">"-"</span>))
    p2 =<span class="st"> </span><span class="kw"><a href="../reference/paramDistBoxplot.html">paramDistBoxplot</a></span>(exampleExoData,<span class="dt">which.param =</span> <span class="st">"beta2"</span>, <span class="dt">names.input =</span> <span class="kw">paste</span>(<span class="st">"Rep"</span>,<span class="dv">1</span>:<span class="dv">3</span>,<span class="dt">sep =</span> <span class="st">"-"</span>))
    gridExtra::<span class="kw">grid.arrange</span>(p1,p2,<span class="dt">nrow =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/param_dist-1.png" width="672" style="display: block; margin: auto;"></p>
<p>Further details over this analysis are in Welch et. al 2016 (submitted). In short, when the ChIP-exo/nexus sample is not deeply sequenced, high values of <span class="math inline">\(\hat{\beta}_1\)</span> indicate that the library complexity is low. In contrast, lower values correspond to higher quality ChIP-exo experiments. We concluded that samples with estimated <span class="math inline">\(\hat{\beta_1} \leq 10\)</span> seem to be high quality samples. Similarly, samples with estimated <span class="math inline">\(\hat{\beta_2} \approx 0\)</span> can be considered as high quality samples. The estimated values for these parameters can be accessed with the <strong>beta1</strong>, <strong>beta2</strong>, and <strong>param_dist</strong> methods. For example, using the <strong>median</strong> to summarize these parameter distributions, we conclude that these three replicates (in <strong>chr1</strong>) are high quality samples:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">sapply</span>(exampleExoData,function(x)<span class="kw">median</span>(<span class="kw"><a href="../reference/beta1.html">beta1</a></span>(x)))</code></pre></div>
<pre><code>## [1] 1.903799 1.498470 8.677156</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    <span class="kw">sapply</span>(exampleExoData,function(x)<span class="kw">median</span>(-<span class="kw"><a href="../reference/beta2.html">beta2</a></span>(x)))</code></pre></div>
<pre><code>## [1] 0.017131301 0.008937054 0.061447391</code></pre>
</div>
<div id="subsampling-reads-from-the-experiment-to-asses-quality" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#subsampling-reads-from-the-experiment-to-asses-quality" class="anchor"> </a></body></html>Subsampling reads from the experiment to asses quality</h2>
<p>The behavior of the third&rsquo;s FoxA1 replicate may be an indication of problems in the sample. However, it is also common to observe that pattern in deeply sequenced experiments. For convenience, we added the function <code>ExoDataSubsampling</code>, that performs the analysis suggested by Welch et. al 2016 (submitted) when the experiment is deeply sequenced. To use this function, we proceed as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    sample.depth =<span class="st"> </span><span class="kw">seq</span>(<span class="fl">1e5</span>,<span class="fl">2e5</span>,<span class="fl">5e4</span>)
    exoList =<span class="st"> </span><span class="kw"><a href="../reference/ExoDataSubsampling.html">ExoDataSubsampling</a></span>(<span class="dt">file =</span> files[<span class="dv">3</span>],<span class="dt">sample.depth =</span> sample.depth, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>The output of <code>ExoDataSubsampling</code> is a list of <code>ExoData</code> objects, therefore its output can be used with any of the plotting functions to asses the quality of the samples. For example, using we may use <code>paramDistBoxplot</code> to get the following figures:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">    p1 =<span class="st"> </span><span class="kw"><a href="../reference/paramDistBoxplot.html">paramDistBoxplot</a></span>(exoList,<span class="dt">which.param =</span> <span class="st">"beta1"</span>)
    p2 =<span class="st"> </span><span class="kw"><a href="../reference/paramDistBoxplot.html">paramDistBoxplot</a></span>(exoList,<span class="dt">which.param =</span> <span class="st">"beta2"</span>)
    gridExtra::<span class="kw">grid.arrange</span>(p1,p2,<span class="dt">nrow =</span> <span class="dv">1</span>)</code></pre></div>
<p><img src="vignette_files/figure-html/subsamplots-1.png" width="672" style="display: block; margin: auto;"></p>
<p>Clearly there are increasing trends in both plots, and since we are only using the reads in chromosome 1, we are observing fewer reads than in a typical ChIP-exo/nexus experiment. In a higher quality experiment it is expected to show lower <span class="math inline">\(\hat{\beta}_1\)</span> and <span class="math inline">\(\hat{\beta}_2\)</span> levels. Additionally, the rate at which the estimated <span class="math inline">\(\hat{\beta}_2\)</span> parameter increases is going to be higher in a lower quality experiment.</p>
</div>
</div>
<div id="conclusions" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#conclusions" class="anchor"> </a></body></html>Conclusions</h1>
<p>We presented a systematic exploration of a ChIP-exo experiment and show how to use the QC pipeline provided in <em><a href="http://bioconductor.org/packages/ChIPexoQual">ChIPexoQual</a></em>. ChIPexoQual takes aligned reads as input and automatically generates several diagnostic plots and summary measures that enable assessing enrichment and library complexity. The implications of the diagnostic plots and the summary measures align well with more elaborate analysis that is computationally more expensive to perform and/or requires additional imputes that often may not be available.</p>
</div>
<div id="sessioninfo" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#sessioninfo" class="anchor"> </a></body></html>SessionInfo</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>(<span class="st">"ChIPexoQual"</span>)</code></pre></div>
<pre><code>## R version 3.3.2 (2016-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 14.04.4 LTS
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## character(0)
## 
## other attached packages:
## [1] ChIPexoQual_0.99.13
## 
## loaded via a namespace (and not attached):
##   [1] utils_3.3.2                   nlme_3.1-128                 
##   [3] bitops_1.0-6                  devtools_1.12.0.9000         
##   [5] RColorBrewer_1.1-2            httr_1.2.1                   
##   [7] highlight_0.4.7               rprojroot_1.1                
##   [9] GenomeInfoDb_1.10.1           tools_3.3.2                  
##  [11] backports_1.0.4               R6_2.2.0                     
##  [13] rpart_4.1-10                  Hmisc_4.0-0                  
##  [15] DBI_0.5-1                     lazyeval_0.2.0               
##  [17] BiocGenerics_0.20.0           colorspace_1.3-1             
##  [19] nnet_7.3-12                   withr_1.0.2                  
##  [21] gridExtra_2.2.1               mnormt_1.5-5                 
##  [23] curl_2.3                      graph_1.52.0                 
##  [25] Biobase_2.34.0                BiocCheck_1.10.0             
##  [27] htmlTable_1.7                 datasets_3.3.2               
##  [29] xml2_1.0.0.9001               desc_1.0.1                   
##  [31] labeling_0.3                  rtracklayer_1.34.1           
##  [33] base_3.3.2                    scales_0.4.1                 
##  [35] hexbin_1.27.1                 psych_1.6.9                  
##  [37] RBGL_1.50.0                   callr_1.0.0.9000             
##  [39] pkgdown_0.1.0.9000            stringr_1.1.0                
##  [41] digest_0.6.10                 Rsamtools_1.26.1             
##  [43] foreign_0.8-67                rmarkdown_1.2                
##  [45] XVector_0.14.0                dichromat_2.0-0              
##  [47] htmltools_0.3.5               ensembldb_1.6.2              
##  [49] BSgenome_1.42.0               grDevices_3.3.2              
##  [51] RSQLite_1.1                   ChIPexoQualExample_0.99.3    
##  [53] BiocInstaller_1.24.0          shiny_0.14.2                 
##  [55] jsonlite_1.1                  BiocParallel_1.8.1           
##  [57] acepack_1.4.1                 dplyr_0.5.0                  
##  [59] VariantAnnotation_1.20.2      RCurl_1.95-4.8               
##  [61] magrittr_1.5                  Formula_1.2-1                
##  [63] Matrix_1.2-7.1                Rcpp_0.12.8                  
##  [65] munsell_0.4.3                 S4Vectors_0.12.1             
##  [67] viridis_0.3.4                 whisker_0.3-2                
##  [69] stringi_1.1.2                 yaml_2.1.14                  
##  [71] SummarizedExperiment_1.4.0    zlibbioc_1.20.0              
##  [73] biocViews_1.42.0              pkgbuild_0.0.0.9000          
##  [75] plyr_1.8.4                    AnnotationHub_2.6.4          
##  [77] grid_3.3.2                    parallel_3.3.2               
##  [79] crayon_1.3.2                  methods_3.3.2                
##  [81] lattice_0.20-34               Biostrings_2.42.1            
##  [83] splines_3.3.2                 GenomicFeatures_1.26.0       
##  [85] knitr_1.15.1                  RUnit_0.4.31                 
##  [87] optparse_1.3.2                GenomicRanges_1.26.1         
##  [89] codetools_0.2-15              reshape2_1.4.2               
##  [91] biomaRt_2.30.0                stats4_3.3.2                 
##  [93] pkgload_0.0.0.9000            XML_3.98-1.5                 
##  [95] evaluate_0.10                 biovizBase_1.22.0            
##  [97] latticeExtra_0.6-28           data.table_1.10.0            
##  [99] httpuv_1.3.3                  testthat_1.0.2               
## [101] graphics_3.3.2                purrr_0.2.2                  
## [103] gtable_0.2.0                  getopt_1.20.0                
## [105] tidyr_0.6.0                   assertthat_0.1               
## [107] ggplot2_2.2.0                 mime_0.5                     
## [109] xtable_1.8-2                  broom_0.4.1                  
## [111] roxygen2_5.0.1                survival_2.40-1              
## [113] tibble_1.2                    GenomicAlignments_1.10.0     
## [115] stats_3.3.2                   AnnotationDbi_1.36.0         
## [117] memoise_1.0.0                 IRanges_2.8.1                
## [119] cluster_2.0.5                 interactiveDisplayBase_1.12.0
## [121] BiocStyle_2.2.1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor"><html><body><a href="#references" class="anchor"> </a></body></html>References</h1>
<div id="refs" class="references">
<div id="ref-exoillumina">
<p>Serandour, Aurelien, Brown Gordon, Joshua Cohen, and Jason Carroll. 2013. &ldquo;Development of and Illumina-Based ChIP-Exonuclease Method Provides Insight into FoxA1-DNA Binding Properties.&rdquo; <em>Genome Biology</em>.</p>
</div>
<div id="ref-qcpipeline">
<p>Welch, Rene, Dongjun Chung, Jeffrey Grass, Robert Landick, and S&uuml;nd&uuml;z Kele&#351;. 2016. &ldquo;Data Exploration, Quality Control, and Statistical Analysis of ChIP-Exo/Nexus Experiments.&rdquo; <em>Submitted</em>.</p>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked"><li><a href="#overview">Overview</a></li>
      <li><a href="#creating-an-exodata-object">Creating an <strong>ExoData</strong> object</a><ul class="nav nav-pills nav-stacked"><li><a href="#enrichment-analysis-and-library-complexity">Enrichment analysis and library complexity:</a></li>
      <li><a href="#strand-imbalance">Strand imbalance</a></li>
      <li><a href="#quality-evaluation">Quality evaluation</a></li>
      <li><a href="#subsampling-reads-from-the-experiment-to-asses-quality">Subsampling reads from the experiment to asses quality</a></li>
      </ul></li>
      <li><a href="#conclusions">Conclusions</a></li>
      <li><a href="#sessioninfo">SessionInfo</a></li>
      <li><a href="#references">References</a></li>
      </ul></div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Dongjun Chung, Sunduz Keles, Rene Welch.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer></div>

  </body></html>
