<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>ExoDataSubsampling — ExoDataSubsampling • ChIPexoQual</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>

<!-- mathjax -->
<script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">ChIPexoQual</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>ExoDataSubsampling</h1>
    </div>

    
    <p><code>ExoDataSubsampling</code> samples <code>sample.reads</code> from the ChIP-exo experiment and creates a list
of <code>ExoData</code> objects</p>
    

    <pre><span class='fu'>ExoDataSubsampling</span>(<span class='kw'>file</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>reads</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>sample.depth</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>height</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>nregions</span> <span class='kw'>=</span> <span class='fl'>1000</span>, <span class='kw'>ntimes</span> <span class='kw'>=</span> <span class='fl'>1000</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>save.reads</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>mc.cores</span> <span class='kw'>=</span> <span class='fu'>getOption</span>(<span class='st'>"mc.cores"</span>, <span class='fl'>2L</span>))</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <dl class="dl-horizontal">
      <dt>file</dt>
      <dd>a character value with location of the bam file with the aligned
reads.</dd>
      <dt>reads</dt>
      <dd>a <code>GAlignments</code> object with the aligned reads of a ChIP-exo
sample. It is meant to be used instead of <code>file</code>.</dd>
      <dt>sample.depth</dt>
      <dd>a numeric vector with the number of reads to be sampled.</dd>
      <dt>height</dt>
      <dd>a numeric value indicating the value used to slice the coverage
of the experiment into a set of regions.</dd>
      <dt>nregions</dt>
      <dd>a numeric value indicating the number of regions sampled to 
estimate the quality parameter distributions. The default value is 1e3.</dd>
      <dt>ntimes</dt>
      <dd>a numeric value indicating the number of times that regions are 
sampled to estimate the quality parameter distributions. The default value
is 1e2.</dd>
      <dt>verbose</dt>
      <dd>a logical value indicating if the user want to receive progress
details. The default value is FALSE.</dd>
      <dt>save.reads</dt>
      <dd>a logical value to indicate if the reads are stored in the
<code>ExoData</code> object. The default value is <code>FALSE</code>.</dd>
      <dt>mc.cores</dt>
      <dd>a numeric value with the number of cores to use,
i.e. at most how many child processes will be run simultaneously.</dd>
    </dl>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>It returns an <code>ExoData</code> object with the regions obtained after
partitioning the genome and the summary statistics for each region. If the
<code>save.reads</code> parameter is <code>TRUE</code> then it contains a <code>GRanges</code>
object with the reads of the ChIP-exo experiment.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='no'>files</span> <span class='kw'>&lt;-</span> <span class='fu'>list.files</span>(<span class='fu'>system.file</span>(<span class='st'>"extdata"</span>,<span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"ChIPexoQualExample"</span>),
    <span class='kw'>full.names</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='no'>sample.depth</span> <span class='kw'>&lt;-</span> <span class='fu'>seq</span>(<span class='fl'>1e5</span>,<span class='fl'>2e5</span>,<span class='fl'>5e4</span>)
<span class='fu'>ExoDataSubsampling</span>(<span class='kw'>file</span> <span class='kw'>=</span> <span class='no'>files</span>[<span class='fl'>5</span>],<span class='kw'>sample.depth</span> <span class='kw'>=</span> <span class='no'>sample.depth</span>)</div><div class='output co'>#&gt; <span class='message'>Creating &#39;ExoData&#39; for 100,000 reads</span></div><div class='output co'>#&gt; <span class='message'>Sampling reads.</span></div><div class='output co'>#&gt; <span class='message'>Using &#39;reads&#39; argument</span></div><div class='output co'>#&gt; <span class='message'>Keeping reads in object: No</span></div><div class='output co'>#&gt; <span class='message'>Calculating summary statistics</span></div><div class='output co'>#&gt; <span class='message'>Calculating quality scores distribution</span></div><div class='output co'>#&gt; <span class='message'>Done!</span></div><div class='output co'>#&gt; <span class='message'>Creating &#39;ExoData&#39; for 150,000 reads</span></div><div class='output co'>#&gt; <span class='message'>Sampling reads.</span></div><div class='output co'>#&gt; <span class='message'>Using &#39;reads&#39; argument</span></div><div class='output co'>#&gt; <span class='message'>Keeping reads in object: No</span></div><div class='output co'>#&gt; <span class='message'>Calculating summary statistics</span></div><div class='output co'>#&gt; <span class='message'>Calculating quality scores distribution</span></div><div class='output co'>#&gt; <span class='message'>Done!</span></div><div class='output co'>#&gt; <span class='message'>Creating &#39;ExoData&#39; for 200,000 reads</span></div><div class='output co'>#&gt; <span class='message'>Sampling reads.</span></div><div class='output co'>#&gt; <span class='message'>Using &#39;reads&#39; argument</span></div><div class='output co'>#&gt; <span class='message'>Keeping reads in object: No</span></div><div class='output co'>#&gt; <span class='message'>Calculating summary statistics</span></div><div class='output co'>#&gt; <span class='message'>Calculating quality scores distribution</span></div><div class='output co'>#&gt; <span class='message'>Done!</span></div><div class='output co'>#&gt; $`100,000 reads`
#&gt; ExoData object with 72562 ranges and 11 metadata columns:
#&gt;           seqnames                 ranges strand |  fwdReads  revReads
#&gt;              &lt;Rle&gt;              &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;
#&gt;       [1]     chr1     [3001926, 3001961]      * |         1         0
#&gt;       [2]     chr1     [3002412, 3002447]      * |         0         2
#&gt;       [3]     chr1     [3003987, 3004022]      * |         0         1
#&gt;       [4]     chr1     [3006189, 3006224]      * |         1         0
#&gt;       [5]     chr1     [3014235, 3014270]      * |         1         0
#&gt;       ...      ...                    ...    ... .       ...       ...
#&gt;   [72558]     chr1 [197171309, 197171344]      * |         1         0
#&gt;   [72559]     chr1 [197175350, 197175385]      * |         1         0
#&gt;   [72560]     chr1 [197184957, 197184992]      * |         0         1
#&gt;   [72561]     chr1 [197188843, 197188878]      * |         1         0
#&gt;   [72562]     chr1 [197189679, 197189714]      * |         1         0
#&gt;              fwdPos    revPos     depth uniquePos                ARC       URC
#&gt;           &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;          &lt;numeric&gt; &lt;numeric&gt;
#&gt;       [1]         1         0         1         1 0.0277777777777778         1
#&gt;       [2]         0         1         2         1 0.0555555555555556       0.5
#&gt;       [3]         0         1         1         1 0.0277777777777778         1
#&gt;       [4]         1         0         1         1 0.0277777777777778         1
#&gt;       [5]         1         0         1         1 0.0277777777777778         1
#&gt;       ...       ...       ...       ...       ...                ...       ...
#&gt;   [72558]         1         0         1         1 0.0277777777777778         1
#&gt;   [72559]         1         0         1         1 0.0277777777777778         1
#&gt;   [72560]         0         1         1         1 0.0277777777777778         1
#&gt;   [72561]         1         0         1         1 0.0277777777777778         1
#&gt;   [72562]         1         0         1         1 0.0277777777777778         1
#&gt;                 FSR         M         A
#&gt;           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;       [1]         1      -Inf       Inf
#&gt;       [2]         0      -Inf      -Inf
#&gt;       [3]         0      -Inf      -Inf
#&gt;       [4]         1      -Inf       Inf
#&gt;       [5]         1      -Inf       Inf
#&gt;       ...       ...       ...       ...
#&gt;   [72558]         1      -Inf       Inf
#&gt;   [72559]         1      -Inf       Inf
#&gt;   [72560]         0      -Inf      -Inf
#&gt;   [72561]         1      -Inf       Inf
#&gt;   [72562]         1      -Inf       Inf
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths
#&gt; 
#&gt; $`150,000 reads`
#&gt; ExoData object with 96484 ranges and 11 metadata columns:
#&gt;           seqnames                 ranges strand |  fwdReads  revReads
#&gt;              &lt;Rle&gt;              &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;
#&gt;       [1]     chr1     [3002412, 3002447]      * |         0         1
#&gt;       [2]     chr1     [3014235, 3014270]      * |         1         0
#&gt;       [3]     chr1     [3014692, 3014727]      * |         1         0
#&gt;       [4]     chr1     [3016288, 3016323]      * |         1         0
#&gt;       [5]     chr1     [3016377, 3016412]      * |         1         0
#&gt;       ...      ...                    ...    ... .       ...       ...
#&gt;   [96480]     chr1 [197184957, 197184992]      * |         0         1
#&gt;   [96481]     chr1 [197188843, 197188878]      * |         1         0
#&gt;   [96482]     chr1 [197189055, 197189090]      * |         2         0
#&gt;   [96483]     chr1 [197192288, 197192323]      * |         0         2
#&gt;   [96484]     chr1 [197192797, 197192832]      * |         1         0
#&gt;              fwdPos    revPos     depth uniquePos                ARC       URC
#&gt;           &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;          &lt;numeric&gt; &lt;numeric&gt;
#&gt;       [1]         0         1         1         1 0.0277777777777778         1
#&gt;       [2]         1         0         1         1 0.0277777777777778         1
#&gt;       [3]         1         0         1         1 0.0277777777777778         1
#&gt;       [4]         1         0         1         1 0.0277777777777778         1
#&gt;       [5]         1         0         1         1 0.0277777777777778         1
#&gt;       ...       ...       ...       ...       ...                ...       ...
#&gt;   [96480]         0         1         1         1 0.0277777777777778         1
#&gt;   [96481]         1         0         1         1 0.0277777777777778         1
#&gt;   [96482]         1         0         2         1 0.0555555555555556       0.5
#&gt;   [96483]         0         1         2         1 0.0555555555555556       0.5
#&gt;   [96484]         1         0         1         1 0.0277777777777778         1
#&gt;                 FSR         M         A
#&gt;           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;       [1]         0      -Inf      -Inf
#&gt;       [2]         1      -Inf       Inf
#&gt;       [3]         1      -Inf       Inf
#&gt;       [4]         1      -Inf       Inf
#&gt;       [5]         1      -Inf       Inf
#&gt;       ...       ...       ...       ...
#&gt;   [96480]         0      -Inf      -Inf
#&gt;   [96481]         1      -Inf       Inf
#&gt;   [96482]         1      -Inf       Inf
#&gt;   [96483]         0      -Inf      -Inf
#&gt;   [96484]         1      -Inf       Inf
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths
#&gt; 
#&gt; $`200,000 reads`
#&gt; ExoData object with 115669 ranges and 11 metadata columns:
#&gt;            seqnames                 ranges strand |  fwdReads  revReads
#&gt;               &lt;Rle&gt;              &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;
#&gt;        [1]     chr1     [3002412, 3002447]      * |         0         1
#&gt;        [2]     chr1     [3003987, 3004022]      * |         0         1
#&gt;        [3]     chr1     [3014692, 3014727]      * |         2         0
#&gt;        [4]     chr1     [3015843, 3015878]      * |         0         1
#&gt;        [5]     chr1     [3016288, 3016340]      * |         2         0
#&gt;        ...      ...                    ...    ... .       ...       ...
#&gt;   [115665]     chr1 [197183268, 197183303]      * |         1         0
#&gt;   [115666]     chr1 [197188637, 197188672]      * |         0         1
#&gt;   [115667]     chr1 [197189055, 197189090]      * |         1         0
#&gt;   [115668]     chr1 [197189679, 197189714]      * |         2         0
#&gt;   [115669]     chr1 [197192288, 197192323]      * |         0         1
#&gt;               fwdPos    revPos     depth uniquePos                ARC       URC
#&gt;            &lt;integer&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;          &lt;numeric&gt; &lt;numeric&gt;
#&gt;        [1]         0         1         1         1 0.0277777777777778         1
#&gt;        [2]         0         1         1         1 0.0277777777777778         1
#&gt;        [3]         1         0         2         1 0.0555555555555556       0.5
#&gt;        [4]         0         1         1         1 0.0277777777777778         1
#&gt;        [5]         2         0         2         2 0.0377358490566038         1
#&gt;        ...       ...       ...       ...       ...                ...       ...
#&gt;   [115665]         1         0         1         1 0.0277777777777778         1
#&gt;   [115666]         0         1         1         1 0.0277777777777778         1
#&gt;   [115667]         1         0         1         1 0.0277777777777778         1
#&gt;   [115668]         1         0         2         1 0.0555555555555556       0.5
#&gt;   [115669]         0         1         1         1 0.0277777777777778         1
#&gt;                  FSR         M         A
#&gt;            &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
#&gt;        [1]         0      -Inf      -Inf
#&gt;        [2]         0      -Inf      -Inf
#&gt;        [3]         1      -Inf       Inf
#&gt;        [4]         0      -Inf      -Inf
#&gt;        [5]         1      -Inf       Inf
#&gt;        ...       ...       ...       ...
#&gt;   [115665]         1      -Inf       Inf
#&gt;   [115666]         0      -Inf      -Inf
#&gt;   [115667]         1      -Inf       Inf
#&gt;   [115668]         1      -Inf       Inf
#&gt;   [115669]         0      -Inf      -Inf
#&gt;   -------
#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths
#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Dongjun Chung, Sunduz Keles, Rene Welch.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
